js/data.js

```diff
@@ -1,8 +1,21 @@
 /* exported data */
-
 var data = {
   view: 'entry-form',
   entries: [],
   editing: null,
-  nextEntryId: 1
+  nextEntryId: 0
 };
+
+var previousData = localStorage.getItem('data-local-storage');
+if (previousData !== null) {
+  data = JSON.parse(previousData);
+}
+
+function handleData(event) {
+  var dataEntry = JSON.stringify(data);
+  localStorage.setItem('data-local-storage', dataEntry);
+}
+
+window.addEventListener('unload', handleData);
+
+data.nextEntryId = data.entries.length + 2;
```

Comment:
This line should not be necessary, since `data.nextEntryId` should be incremented every time a new entry is saved by the user.